internal/service/resources/collector/utils.go

```diff
@@ -0,0 +1,225 @@
+package collector
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/url"
+	"reflect"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5/pgxpool"
+
+	"github.com/openshift-kni/oran-o2ims/internal/model"
+	"github.com/openshift-kni/oran-o2ims/internal/service"
+	"github.com/openshift-kni/oran-o2ims/internal/service/common/db"
+	"github.com/openshift-kni/oran-o2ims/internal/service/common/utils"
+	"github.com/openshift-kni/oran-o2ims/internal/service/resources/db/models"
+)
+
+// GenericModelConverter converts a db model object into an API object and returns it as an `any`
+// so that it can be used generically since the API model doesn't implement a base interface.
+type GenericModelConverter func(object interface{}) any
+
+// persistObject persists an object to its database table.  If the object does not already have a
+// persisted representation then it is created; otherwise any modified fields are updated in the
+// database tuple.  The function returns both the before and after versions of the object.
+func persistObject[T db.Model](ctx context.Context, db *pgxpool.Pool,
+	object T, uuid uuid.UUID) (*T, *T, error) {
```

Comment:
nit: I'm so used to record now that object confused me for a moment. Perhaps for consistency,  consider changing object by record